enhancements/agent-installer/agent-based-installer.md

```diff
@@ -0,0 +1,279 @@
+
+N/A
+
+### Version Skew Strategy
+
+N/A
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+N/A
+
+#### Support Procedures
+
+N/A
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in
+`Implementation History`.
+
+## Drawbacks
+
+In disconnected environments, users will still need to mirror the release
+payload to an accessible local registry, which must be long-running (i.e. it
+continues to be available once the cluster is up).
+
+Since the installation tools cannot continue running to the completion of
+building the cluster (Node 0 must eventually be rebooted to join the cluster),
+it will necessarily not be possible to use a single API to track the entire
+process of installation.
+
+Despite the fact that RHCOS is an operating system designed exclusive to run
+OpenShift, users will still need to wait for OpenShift to be *installed* into
+it after boot. A total of three boots (which are very slow on baremetal due to
+RAM testing at startup) will still be required for each host to come up as an
+OpenShift node.
+
+## Alternatives
+
+### Run assisted-service outside of the cluster hosts
+
+Rather than running assisted-service on one of the hosts that is to become part
+of the cluster, we could require the user to provide somewhere else to run it
+in the same network environment, in either a Podman pod on a RHEL host or in a
+virtual machine. This avoids the complexity of running the installation
+components on a host that will itself have to be rebooted and come up as a
+member of the cluster control plane.
+
+However, there exist environments where the boot image is transported in on
+physical media and the user has no access to any other compute capacity within
+the same network environment. This architecture would preclude supporting such
+sites at any point in the future.
+
+### Expand from a single-node cluster
+
+Instead of doing an 'installation' as such, we could provide a pre-built
+Single-Node OpenShift image with the Multi-Cluster Engine (and potentially a
+registry with a mirror of the release image) installed. Users could then use
+MCE to expand the cluster to the desired size.
+
+However, scaling etcd from 1 to multiple nodes carries a significant risk of
+data loss (unlike scaling from 3 to 5 or more nodes). While this is likely
+acceptable during installation, it is not at any other time and there are thus
+no plans to implement this scenario in the etcd operator.
+
+### Deliver the assisted components separately
+
+Rather than delivering the assisted components as part of the release image, we
+could ship them separately and have the ISO pull them either directly from MCE
+or from an operator in the OLM catalog. In disconnected environments, this
+would require the user to perform an extra step to mirror an additional image
+obtained in a different way from the release image.
+
+This would also permit installation of older versions of OpenShift, at least
```

Comment:
Theoretically, and practiallcaly, u can always use the same iso as a discovery entry point - anyhow the os-content is overwritten by MCO and there are no optimizations there. 
Few challenges are ignition version (that might be not backwards compatible) and drivers difference between the live iso and rhcos node os

Reply:
I agree, but we have to get the ISO from somewhere. Currently it's in the release payload, so that's the obvious place because there are no extra steps for the user.